Resources/public/js/views/services/plugins/ez-dashboardblockallcontentserviceplugin.js

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-dashboardblockallcontentserviceplugin', function (Y) {
+    'use strict';
+
+    /**
+     * Provides the all content dashboard block service plugin class
+     *
+     * @module ez-dashboardblockallcontentserviceplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    var DEFAULT_ERROR_IDENTIFIER = 'ez-dashboardblockallcontentserviceplugin-error',
+        DEFAULT_ERROR_MESSAGE = 'DashboardBlockAllContentService: error';
+
+    /**
+     * Provides the all content dashboard block service plugin
+     *
+     * @namespace eZ.Plugin
+     * @class DashboardBlockAllContentService
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.DashboardBlockAllContentService = Y.Base.create('dashboardBlockAllContentServicePlugin', Y.eZ.Plugin.ViewServiceBase, [], {
+        initializer: function () {
+            this.onHostEvent('*:getAllContent', this._getAllContent, this);
+            this.onHostEvent('*:gotoContentEdit', this._goToContentEdit, this);
+            this.onHostEvent('*:gotoContentPreview', this._goToContentPreview, this);
+        },
+
+        /**
+         * Gets all content for the all content dashboard block view
+         *
+         * @method _getAllContent
+         * @protected
+         * @param event {Object} event facade
+         */
+        _getAllContent: function (event) {
+            this._loadRootLocation()
+                .then(Y.bind(this._loadRootLocationModelData, this))
+                .then(Y.bind(this._getContentList, this, event.target))
+                .catch(Y.bind(this._handleError, this));
+        },
+
+        /**
+         * Gets a list of content
+         *
+         * @method _getContentList
+         * @protected
+         * @param target {eZ.DashboardBlockAllContentView} a view requesting a content
+         */
+        _getContentList: function (target) {
+            this.get('host').fire('contentSearch', {
+                viewName: 'all-content',
+                loadContentType: true,
+                search: {
+                    criteria: {
+                        SubtreeCriterion: this.get('rootLocation').get('pathString')
+                    },
+                    sortClauses: {
```

Comment:
just one thing on that: this is not yet supported server side. Meaning the block will not display the modified content but a *random* list of content item. See https://jira.ez.no/browse/EZP-24315 (added it as a blocker).